src/libraries/System.Diagnostics.Process/tests/ProcessStandardConsoleTests.cs

```diff
@@ -11,9 +11,14 @@ namespace System.Diagnostics.Tests
     public class ProcessStandardConsoleTests : ProcessTestBase
     {
         [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
-        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer), nameof(PlatformDetection.IsNotWindowsServerCore), nameof(PlatformDetection.IsNotWindowsIoTCore))]
         public void TestChangesInConsoleEncoding()
         {
+            // These OS variants don't work with setting a default Encoding.
+            if (PlatformDetection.IsNotWindowsNanoServer || PlatformDetection.IsNotWindowsServerCore || PlatformDetection.IsNotWindowsIoTCore)
```

Comment:
This should not be needed as I fixed the `Platform.IsInContainer` check for windows containers in: https://github.com/dotnet/runtime/pull/41701 -- so the check below doing: `if (!OperatingSystem.IsWindows() || PlatformDetection.IsInContainer)` should handle the container case correctly.

Reply:
Cool, thanks.

Reply:
Unrelated, but won't this still fail if Windows Nano Server isn't run inside a container?

Reply:
Ah, yeah. It would fail if it is not in a container. We don't run them outside a container on those OSs, but maybe we can add IsWindowsNanoServer to the condition below?

Reply:
are we only concerned about nano server? what about other OSs like Server Core or IOT Server? Unsure if they still exist.

Reply:
I will leave this as is as it's unrelated but we might want to do a local pass at some point...